### DISCUSSION TITLE
Adapting open web ui pipelines for different services

### DISCUSSION THREAD

**Discussion by sir3mat** [open-webui/pipelines#184] [General]
Hi, I would like to develop a pipeline for tracing using Phoenix or MLflow. How can I add custom dependencies with the Helm chart? Do I need to package a dedicated image starting from the pipeline image, or are there different approaches, such as deploying my own pipeline server with the custom dependencies? Alternatively, do I need to use a general pipeline that sends requests to another custom API that wraps the Phoenix or MLflow logic? This could be one solution, but it might be a bit overwhelming

**Comment by PlebeiusGaragicus** [comment]
My solution is to keep my pipeline very very simple - it just calls another endpoint.

That second endpoint is my own FastAPI server that ALL my business and ML logic goes into.  I've tried it various ways, and I think this is best.  The only downside is that you have multiple servers running all working side by side - which can be a lot to manage.

I suppose you could fork the pipelines repo and put ALL your logic in there, but I found that too brittle.
